stdlib/src/builtin/string.mojo

```diff
@@ -598,6 +598,34 @@ fn _isspace(c: UInt8) -> Bool:
     return _SPACES_TABLE[int(c)]
 
 
+# ===----------------------------------------------------------------------=== #
+# _isnewline
+# ===----------------------------------------------------------------------=== #
+
+
+fn _get_newlines_table() -> InlineArray[UInt8, 128]:
+    var table = InlineArray[UInt8, 128](0)
+    table[ord("\n")] = 1
+    table[ord("\r")] = 1
+    table[ord("\f")] = 1
+    table[ord("\v")] = 1
+    table[ord("\x1c")] = 1
+    table[ord("\x1d")] = 1
+    table[ord("\x1e")] = 1
+    return table
+
+
+alias _NEWLINES_TABLE = _get_newlines_table()
+
+
+fn _isnewline(c: String) -> Bool:
+    # TODO: add \u2028 and \u2029 when they are properly parsed
+    # FIXME: \x85 is parsed but not encoded in utf-8
```

Comment:
I haven't looked, but isn't this a problem with the implementation of `ord`?

Reply:
No. `ord` is correctly implemented as far as I can tell. It recognize unicode characters.
```mojo
  3> var s = String("\xc2\x85"); 
(String) s = "\U00000085"
  4> print(s._buffer.__str__()) 
  5.  
[SIMD[DType.uint8, 1](194), SIMD[DType.uint8, 1](133), SIMD[DType.uint8, 1](0)]
  
5> print(ord(s))
133

```

The problem is that `\x85` *itself* is not encoded in utf-8 but in ASCII.

```
   6> var s = String("\x85")
(String) s = "\x85"
  7> print(s._buffer.__str__())
[SIMD[DType.uint8, 1](133), SIMD[DType.uint8, 1](0)]

```

For the some reason, `\u2028` is not recognized 
```
  8> var s = String("\u2028")
[User] error: Expression [7]:1:17: invalid escape sequence
var s = String("\u2028")
                ^

[User] error: Expression [7]:1:17: invalid escape sequence
var s = String("\u2028")
                ^

[User] error: Expression [7]:1:17: invalid escape sequence
var s = String("\u2028")
                ^

[User] error: Expression [7]:1:17: invalid escape sequence
var s = String("\u2028")
                ^

```

The is probably a bug in mojo's parser.

Reply:
`ord("\x85")` treats "\x85" as a unicode character "\xc2\x85" but in fact it is in ASCII hence `ord("\x85")==5`

Reply:
I opened an issue about it 3 days ago #2842

> ord("\x85") treats "\x85" as a unicode character

that is how python does it and \x85 is the way that it is annotated even though as you said it has a preceding \xc2 byte

Reply:
Yeah definitely a bug.